File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/CastExpressionsTest.java
/**
 * Copyright (c) 2014 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */

package org.yakindu.sct.generator.java.test;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;
import org.yakindu.scr.castexpressions.CastExpressionsStatemachine;
import org.yakindu.scr.castexpressions.CastExpressionsStatemachine.State;
/**
 *  Unit TestCase for CastExpressions
 */
@SuppressWarnings("all")
public class CastExpressionsTest {

	private CastExpressionsStatemachine statemachine;

	@Before
	public void setUp() {
		statemachine = new CastExpressionsStatemachine();
		statemachine.init();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}

	@Test
	public void testCastExpressionTest() {
		statemachine.enter();
		assertTrue(statemachine.getRealValue() == 5);
		assertTrue(statemachine.getIntValue() == 5);
		statemachine.runCycle();
		assertTrue(statemachine.getRealValue() == 15);
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_C));
		assertTrue(statemachine.getRealValue() == 757);
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/CastExpressions/CastExpressionsTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "CastExpressions.h"

static CastExpressions statechart;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		castExpressions_init(&statechart);
	}
};


TEST_F(StatemachineTest, CastExpressionTest) {					
	castExpressions_enter(&statechart);
	EXPECT_TRUE(castExpressionsIface_get_realValue(&statechart)== 5l);
	EXPECT_TRUE(castExpressionsIface_get_intValue(&statechart)== 5l);
	castExpressions_runCycle(&statechart);
	EXPECT_TRUE(castExpressionsIface_get_realValue(&statechart)== 15l);
	castExpressions_runCycle(&statechart);
	EXPECT_TRUE(castExpressions_isStateActive(&statechart, CastExpressions_main_region_C));
	EXPECT_TRUE(castExpressionsIface_get_realValue(&statechart)== 757l);
}


